server/auto_updater/dataset_updater.py

```diff
@@ -0,0 +1,305 @@
+
+    # skips weird documents that don't contain text
+    if len(word_vectors) > 0:
+        return np.stack(word_vectors).mean(axis=0), Counter(all_tokens)
+
+    return [], None
+
+
+def merge_files(current_file, temp_file):
+    """
+    This function is designed to append entries from the temporary file into the file
+    that needs to be updated
+
+    Parameters:
+        current_file - the file to be updated
+        temp_file - the file that conatins information on new papers
+    """
+
+    if Path(current_file).suffix == ".xz":
+        outfile = lzma.open(current_file, "at")
+
+    else:
+        outfile = open(current_file, "a")
+
+    infile = open(temp_file, "r")
+    inreader = csv.DictReader(infile, delimiter="\t")
+    outwriter = None
+
+    for line in tqdm.tqdm(inreader):
+        fieldnames = line.keys()
+
+        if outwriter is None:
+            outwriter = csv.DictWriter(outfile, delimiter="\t", fieldnames=fieldnames)
+
+        outwriter.writerow(line)
+
+    infile.close()
+    outfile.close()
+
+
+def update_dictionaries(global_word_obj, paper_landscape_file, new_word_counts):
+    """
+    This function is designed to update the current token counts used for the preprint landscape.
+
+    Parameters:
+        global_word_obj - the file path for the background count pickle file
+        paper_landscape_file - the file that contains the bins each paper is assigned to
+        new_word_counts - The file that contains token counts for all new papers
+    """
+
+    all_paper_bins = pd.read_csv(paper_landscape_file, sep="\t")
+    bin_dict = dict(
+        zip(
+            all_paper_bins["document"].tolist(), all_paper_bins["squarebin_id"].tolist()
+        )
+    )
+    with open(global_word_obj, "rb") as word_count_file:
+        global_word_counter_obj = pickle.load(word_count_file)
+
+    token_bin_dictionaries = defaultdict(Counter)
+
+    with open(new_word_counts, "r") as infile:
+        count_reader = csv.DictReader(
+            infile,
+            delimiter="\t",
+        )
+
+        for line in tqdm.tqdm(count_reader):
+            square_bin = bin_dict[line["document"]]
+            token_bin_dictionaries[square_bin].update(
+                {line["lemma"]: int(line["count"])}
+            )
+            global_word_counter_obj[line["lemma"]] += int(line["count"])
+
+    with open(global_word_obj, "wb") as word_count_file:
+        pickle.dump(global_word_counter_obj, word_count_file)
+
+    # Get max number of characters
+    max_num = len(str(max(all_paper_bins["squarebin_id"].tolist())))
+
+    # For each bin update the token count
+    for square_bin in token_bin_dictionaries:
+        bin_num_str = "0" * (max_num - len(str(square_bin)))
+
+        pkl_filename = (
+            f"bin_counters/word_bin_{bin_num_str + str(square_bin)}_count.pkl"
+        )
+        if Path(pkl_filename).exists():
+            object_to_be_updated = pickle.load(open(pkl_filename, "rb"))
```

Comment:
@danich1: After checking the new pickle files in `bin_counters/` folder, I feel that it may not be correct to use the original pickle files in `bin_counters/` here, because a lot of papers in original `paper_dataset_tsne_square.tsv` have been assigned to a new `squarebin_id` when `paper_dataset_tsne_square.tsv`  is updated. 
- Example 1: In the former tsne file, `PMC89010` was in squarebin **964**, but in the latter tsne file, it's in squarebin **971**.  
- Example 2: The original squarebin_id of `PMC6207739` was 1475, but was updated to 1490. Because this paper is not new, it's not never included in the updated `word_bin_1490_count.pkl`, but is still incorrectly kept in `word_bin_1475_count.pkl` after updating.

So I think we need to recalculate each squarebin's pickle file based on *ALL* papers (instead of updating the pickle files based on *new* papers only).